ncbi/src/main/java/org/reactome/release/NCBIGene.java

```diff
@@ -0,0 +1,184 @@
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+
+import static org.reactome.release.Utilities.appendWithNewLine;
+
+/**
+ * File generator for NCBI Gene.  This class has logic for producing a file for
+ * NCBI Gene XML, describing the relationship between NCBI Gene identifiers and
+ * Reactome UniProt entries as well as their top level pathways.
+ * It also can create a file for NCBI Protein, describing the relationship between
+ * NCBI Gene identifiers and Reactome UniProt entries in a simple tab delimited format.
+ * @author jweiser
+ */
+public class NCBIGene {
+	private static final Logger logger = LogManager.getLogger();
+	private static final Logger ncbiGeneLogger = LogManager.getLogger("ncbiGeneLog");
+
+	private static final String rootTag = "LinkSet";
+
+	private List<NCBIEntry> ncbiEntries;
+	private String outputDir;
+	private int reactomeVersion;
+
+	public static NCBIGene getInstance(List<NCBIEntry> ncbiEntries, String outputDir, int reactomeVersion) {
+		return new NCBIGene(ncbiEntries, outputDir, reactomeVersion);
+	}
+
+	private NCBIGene(List<NCBIEntry> ncbiEntries, String outputDir, int reactomeVersion) {
+		this.ncbiEntries = ncbiEntries;
+		this.outputDir = outputDir;
+		this.reactomeVersion = reactomeVersion;
+	}
+
+	/**
+	 * Writes NCBI Protein tab-delimited file describing the UniProt to NCBI Gene identifier relationships in
+	 * Reactome to a pre-set output directory
+	 * @throws IOException Thrown if creating or appending for file fails
+	 */
+	public void writeProteinFile() throws IOException {
+		Path filePath = getProteinFilePath();
+
+		Files.deleteIfExists(filePath);
+		Files.createFile(filePath);
+
+		logger.info("Writing proteins_version file");
+
+		// Write file header
+		Files.write(
+			filePath,
+			"UniProt ID\tGene id"
+				.concat(System.lineSeparator())
+				.concat(System.lineSeparator())
+				.getBytes()
+		);
+
+		// Append map contents
+		Set<String> proteinLines = new LinkedHashSet<>();
+		for (NCBIEntry ncbiEntry : ncbiEntries) {
+			for (String ncbiGeneId : ncbiEntry.getNcbiGeneIds()) {
+				proteinLines.add(ncbiEntry.getUniprotAccession() + "\t" + ncbiGeneId);
+			}
+		}
+
+		for (String line : proteinLines) {
+			appendWithNewLine(line, filePath);
+		}
+
+		logger.info("Finished writing proteins_version file");
+	}
+
+	/**
+	 * Writes NCBI Gene XML files describing the relationships between NCBI Gene identifiers and UniProt entries as
+	 * well as their Reactome pathways to pre-set output directory
+	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
+	 * @param numGeneXMLFiles Number of files to divide the Gene XML entries among
+	 * @throws IOException Thrown if creating or appending for any file fails
+	 */
+	public void writeGeneXMLFiles(Session graphDBSession, int numGeneXMLFiles) throws IOException {
+		Path geneErrorFilePath = getGeneErrorFilePath();
+		Files.deleteIfExists(geneErrorFilePath);
```

Comment:
A potentially trivial suggestion: In the idea of a 'Utilities' class there could be a `deleteAndCreateFile()' method since I've seen it appear multiple times

Reply:
Done!